client/snap_op.go

```diff
@@ -192,23 +213,29 @@ func sendSnapFile(snapPath string, snapFile *os.File, pw *io.PipeWriter, mw *mul
 	if action.SnapOptions == nil {
 		action.SnapOptions = &SnapOptions{}
 	}
-	errs := []error{
-		mw.WriteField("action", action.Action),
-		mw.WriteField("name", action.Name),
-		mw.WriteField("snap-path", action.SnapPath),
-		mw.WriteField("channel", action.Channel),
-		mw.WriteField("devmode", strconv.FormatBool(action.DevMode)),
-		mw.WriteField("jailmode", strconv.FormatBool(action.JailMode)),
-		mw.WriteField("classic", strconv.FormatBool(action.Classic)),
-		mw.WriteField("dangerous", strconv.FormatBool(action.Dangerous)),
-	}
-	for _, err := range errs {
-		if err != nil {
+	for _, s := range []struct {
+		name  string
+		value string
+	}{
+		{"action", action.Action},
+		{"name", action.Name},
+		{"snap-path", action.SnapPath},
+		{"channel", action.Channel},
+	} {
```

Comment:
Same.. pulling the `for` here nicely separates out the definition from the loop.